### PR TITLE
Profile Feeds & global Feeds 

### PR DESCRIPTION
**Pull request by Ebube111**

### CODE REVIEW

File: src/pages/profile/[userId]/feed/index.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { ReactElement } from "react";
+
+import { useRouter } from "next/router";
+
+import { ProfileLayout } from "@/modules/profile";
+import ProfileFeeds from "@/modules/profile/components/ProfileFeeds";
+
+const ProfileFeedsTab = () => {
+  const router = useRouter();
+  const { userId: userIdPathParam } = router.query;
```
Comment (carina-akaia) [inline src/pages/profile/[userId]/feed/index.tsx:10]: Hint: you can use `useRouteQuery` for convenience ( even if you don't need to mutate the search params )

File: src/pages/feed/index.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        alt="pots"
+      />
+    </div>
+  );
+
+  return (
+    <div className="h-full max-h-full w-full max-w-[1300px] overflow-auto p-8">
+      <div className="space-y-4">
+        {feedPosts.map((post, index) => (
+          <FeedCard key={index} post={post} />
```
Comment (carina-akaia) [inline src/pages/feed/index.tsx:114]: [Don't use array indexes as keys](https://react.dev/learn/rendering-lists#where-to-get-your-key)

File: src/modules/profile/components/ProfileFeeds.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+import Post from "./FeedCard";
+
+const ProfileFeeds: React.FC<ProfileFeedsProps> = ({ accountId }) => {
+  const [posts, setPosts] = useState<IndexPostResultItem[]>([]);
+  const [offset, setOffset] = useState(40);
+  const [isLoading, setIsLoading] = useState(false);
+  const loadingRef = useRef<HTMLDivElement | null>(null);
+
+  const client = new Social({
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/profile/components/ProfileFeeds.tsx:21]: resolved

File: src/common/lib/blockHeightToTime.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+export const fetchTimeByBlockHeight = async (
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/common/lib/blockHeightToTime.ts:1]: resolved

File: src/modules/profile/components/FeedCard.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+          }}
+        >
+          {post.content}
+        </ReactMarkdown>
+      </div>
+    </div>
+  );
+};
+
+export default FeedCard;
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/profile/components/FeedCard.tsx:113]: resolved

File: src/modules/profile/components/FeedCard.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+      const time = await fetchTimeByBlockHeight(Number(post.blockHeight));
+      setTime(time);
+      setProfileImg(image);
+    };
+    if (post.accountId) fetchProfileImage();
+  }, [post.accountId]);
+
+  return (
+    <div
+      onClick={() => router.push(`/feed/${post.accountId}/${post.blockHeight}`)}
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/profile/components/FeedCard.tsx:44]: resolved

File: src/pages/feed/[account]/[block]/index.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
+/* eslint-disable @next/next/no-img-element */
+
+"use client";
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/pages/feed/[account]/[block]/index.tsx:3]: resolved

File: src/pages/feed/[account]/[block]/index.tsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+  const client = new Social({
+    contractId: process.env.NEXT_PUBLIC_SOCIAL_DB_CONTRACT_ID,
+    network: process.env.NEXT_PUBLIC_NETWORK,
+  });
+
+  useEffect(() => {
+    const fetchPost = async () => {
+      if (account && block) {
+        setIsLoading(true);
+        try {
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/pages/feed/[account]/[block]/index.tsx:40]: resolved

File: src/modules/profile/components/FeedCardOptionsSelect.tsx
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/profile/components/FeedCardOptionsSelect.tsx:1]: resolved

File: src/pages/feed/index.tsx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+    contractId: process.env.NEXT_PUBLIC_SOCIAL_DB_CONTRACT_ID,
+    network: process.env.NEXT_PUBLIC_NETWORK,
+  });
+
+  useEffect(() => {
+    const fetchRegistrationsAndFeeds = async () => {
+      setLoading(true);
+      try {
+        // Fetch registrations
+        const registrations = await getRegistrations();
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/pages/feed/index.tsx:32]: resolved

File: src/modules/profile/components/ProfileFeeds.tsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+        src="https://ipfs.near.social/ipfs/bafkreibcjfkv5v2e2n3iuaaaxearps2xgjpc6jmuam5tpouvi76tvfr2de"
+        alt="pots"
+      />
+    </div>
+  );
+
+  return (
+    <div className="my-8 h-full max-h-80 w-full">
+      {posts.map((post, index) => (
+        <Post key={index} post={post} />
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/profile/components/ProfileFeeds.tsx:92]: resolved

File: src/pages/profile/[userId]/feed/index.tsx (hunk 1 above)
Comment (Ebube111) [reply src/pages/profile/[userId]/feed/index.tsx:10]: resolved

File: src/pages/feed/index.tsx (hunk 2 above)
Comment (Ebube111) [reply src/pages/feed/index.tsx:114]: resolved

File: src/common/contracts/potlock/interfaces/post.interfaces.ts
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/common/contracts/potlock/interfaces/post.interfaces.ts:1]: resolved

File: src/app/layout.tsx
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/app/layout.tsx:1]: Noted

File: src/pages/feed/index.tsx [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+  const loadingRef = useRef<HTMLDivElement | null>(null);
+  const [offset, setOffset] = useState(40);
+
+  const { data, isLoading } = potlock.useListRegistrations({
+    listId: POTLOCK_REGISTRY_LIST_ID,
+  });
+
+  useEffect(() => {
+    const fetchRegistrationsAndFeeds = () => {
```
Comment (carina-akaia) [inline src/pages/feed/index.tsx:25]:
```suggestion

```

File: src/pages/feed/index.tsx [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+          setFeedPosts(posts); // Flatten the array if necessary
+          setLoading(isLoading);
+        })
+        .catch((err) => {
+          console.error("Error fetching registrations or feeds:", err);
+        })
+        .finally(() => {
+          setLoading(isLoading); // Ensure loading state is updated
+        });
+    };
```
Comment (carina-akaia) [inline src/pages/feed/index.tsx:47]:
```suggestion

```

File: src/pages/feed/index.tsx [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+
+    fetchRegistrationsAndFeeds();
```
Comment (carina-akaia) [inline src/pages/feed/index.tsx:49]:
```suggestion

```

File: src/pages/feed/index.tsx [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+  const [loadingMore, setLoadingMore] = useState<boolean>(false);
+
+  const loadingRef = useRef<HTMLDivElement | null>(null);
+  const [offset, setOffset] = useState(40);
+
+  const { data, isLoading } = potlock.useListRegistrations({
+    listId: POTLOCK_REGISTRY_LIST_ID,
+  });
+
+  useEffect(() => {
```
Comment (carina-akaia) [inline src/pages/feed/index.tsx:24]: You don't need to wrap this effect in a function ( see change suggestions )

Same for [src/pages/feed/[account]/[block]/index.tsx](https://github.com/PotLock/potlock-nextjs-app/pull/132/files#diff-589c0812f677fa064e30afce6f8d886f04cfafbc7d4ba050184195f113172e29)

**Review by carina-akaia — APPROVED** [review]

File: src/app/layout.tsx
[the remark this answers is not in the document]
Comment (carina-akaia) [reply src/app/layout.tsx:1]: @Ebube111 Please don't forget to delete the file

**Comment by carina-akaia** [comment]
Left final comments, other than that good job!

File: src/pages/feed/index.tsx (hunk 12 above)
Comment (Ebube111) [reply src/pages/feed/index.tsx:47]: resolved

**Pull request merged**